### PR TITLE
IDEMPIERE-5850 - Validate Source of Consolidated Documents

### PR DESCRIPTION
**Pull request by PeterTakacs300**
JIRA ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5850

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [ ] I have made corresponding changes to the documentation as follows:
  - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
  - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

File: org.adempiere.base/src/org/compiere/model/MInOut.java [hunk 1]
```diff
@@ -1523,6 +1524,58 @@ public String prepareIt()
...
+			
+			HashSet<Integer> temp_C_Order_ID = new HashSet<Integer>(); // Load C_Order_IDs into Hash Set
+			
+			for(MInOutLine line : getLines())	{
+				
+				if(line.getC_OrderLine_ID() == 0) {
+					temp_C_Order_ID.add(0); // Add zero C_Order_ID
+				}
+				else {
+					temp_C_Order_ID.add(line.getC_OrderLine().getC_Order_ID());
```
Comment (CarlosRuiz-globalqss) [inline org.adempiere.base/src/org/compiere/model/MInOut.java:1538]: The call to getC_OrderLine() is very expensive.

I think this code can be made a lot simpler like this:
```java
if (getC_Order_ID() > 0) {
    int[] orderIds = DB.getIDsEx(get_TrxName(), "SELECT DISTINCT C_Order_ID FROM C_OrderLine WHERE C_Order_ID=?", getC_Order_ID());
    if (orderIds.length == 1 && orderIds[0] != getC_Order_ID())
        setC_Order_ID(orderIds[0]);
    else if (orderIds.length > 1)
        setC_Order_ID(0);
}
```

**Review by CarlosRuiz-globalqss — CHANGES_REQUESTED** [review]
Please check the suggested improvement.

File: org.adempiere.base/src/org/compiere/model/MInOut.java (hunk 1 above)
Comment (PeterTakacs300) [reply org.adempiere.base/src/org/compiere/model/MInOut.java:1538]: Hi @CarlosRuiz-globalqss - but in this case, wouldn't `SELECT DISTINCT C_Order_ID FROM C_OrderLine WHERE C_Order_ID=?` return always the C_Order_ID set on our M_InOut?

The point would be to check if M_InOutLine -> C_OrderLine -> C_Order_ID == M_InOut.C_Order_ID, I mean something like this:
`SELECT DISTINCT ol.C_Order_ID FROM M_InOutLine iol JOIN C_OrderLine ol ON (iol.C_OrderLine_ID=ol.C_OrderLine_ID) WHERE iol.M_InOut_ID=?`

Or am I missing something?

Comment (PeterTakacs300) [reply org.adempiere.base/src/org/compiere/model/MInOut.java]: Thanks @CarlosRuiz-globalqss, I've committed the changes. With this we removed 41 lines of code and the behaviour is the same :)

**Comment by PeterTakacs300** [comment]
Hi @CarlosRuiz-globalqss, are there any further changes necessary to this PR?

**Review by CarlosRuiz-globalqss — APPROVED** [review]
tested

**Pull request merged**